platforms/core-configuration/configuration-cache/src/crossVersionTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheProblemsCrossVersionTest.groovy

```diff
@@ -28,7 +28,7 @@ import org.gradle.tooling.events.problems.Severity
 import org.gradle.tooling.events.problems.SingleProblemEvent
 import org.gradle.tooling.events.problems.internal.GeneralData
 
-@ToolingApiVersion(">=8.10")
+@ToolingApiVersion(">=8.10 <8.12")
```

Comment:
I extracted a common `Problem` interface on the Tooling API client side, that is being used in `org.gradle.tooling.Failure` and in `org.gradle.tooling.events.problems.SingleProblemEvent`. This is a breaking change which required adjustments in the cross-version coverage. I've kept the existing tests with an upper bound then copied and adjusted them in the `r812` package.